mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -3043,116 +3199,15 @@ LogicalResult OpenMPDialectLLVMIRTranslationInterface::convertOperation(
     LLVM::ModuleTranslation &moduleTranslation) const {
 
   llvm::OpenMPIRBuilder *ompBuilder = moduleTranslation.getOpenMPBuilder();
+  if (ompBuilder->Config.isTargetDevice()) {
+    if (isInternalTargetDeviceOp(op)) {
+      return convertInternalTargetOp(op, builder, moduleTranslation);
```

Comment:
Nit: Can't we remove `convertInternalTargetOp` and simplify logic here a bit?
```c++
if (ompBuilder->Config.isTargetDevice() && !isTargetDeviceOp(op))
  return convertNestedTargetOps(...);
return convertOperation(...);
```

Reply:
We can, but we'd have to re-introduce this kind of logic in a subsequent patch anyway, since we're going to have specialized lowering on the target side (distribute parallel wsloop reductions) that does not apply to the host codegen. I was thinking it might be better to put in the more permanent code structure right away, but if you have strong opinions about it we can change it.

Reply:
I don't quite have the full picture in my head at the moment, but I think we can leave it as you proposed. When that patch comes maybe I'll be convinced that it is the right way to go, or we could discuss further at that time.

One thing I'm still struggling to understand is the use of "internal" and "common" in function names. If these are used to identify device-only vs host+device conversions, maybe it would make sense that names would reflect that difference. So `convertInternalTargetOp` would be `convertDeviceTargetOp` and `convertCommonOperation` would be `convertHostOrDeviceOperation` or similar. Maybe I misunderstood their meaning, in which case maybe a name change or additional documentation might still be good to have.